Lesson4/step2/project.py

```diff
@@ -251,209 +243,207 @@ def gdisconnect():
-    else:
-        return render_template('menu.html', items=items, restaurant=restaurant, creator=creator)
+    return render_template('menu.html', items=items, restaurant=restaurant, creator=creator,
+                           login_session=login_session)
 
 
 # Create a new menu item
 @app.route('/restaurant/<int:restaurant_id>/menu/new/', methods=['GET', 'POST'])
-def newMenuItem(restaurant_id):
+def new_menu_item(restaurant_id):
     if 'username' not in login_session:
         return redirect('/login')
     restaurant = session.query(Restaurant).filter_by(id=restaurant_id).one()
     if login_session['user_id'] != restaurant.user_id:
-        return "<script>function myFunction() {alert('You are not authorized to add menu items to this restaurant. Please create your own restaurant in order to add items.');}</script><body onload='myFunction()''>"
-        if request.method == 'POST':
-            newItem = MenuItem(name=request.form['name'], description=request.form['description'], price=request.form[
-                               'price'], course=request.form['course'], restaurant_id=restaurant_id, user_id=restaurant.user_id)
-            session.add(newItem)
-            session.commit()
-            flash('New Menu %s Item Successfully Created' % (newItem.name))
-            return redirect(url_for('showMenu', restaurant_id=restaurant_id))
+        return "<script>function myFunction() {" \
+               "alert('You are not authorized to add menu items to this restaurant. " \
+               "Please create your own restaurant in order to add items.');}</script><body onload='myFunction()''>"
+    if request.method == 'POST':
+        new_item = MenuItem(name=request.form['name'], description=request.form['description'], price=request.form[
+            'price'], course=request.form['course'], restaurant_id=restaurant_id, user_id=restaurant.user_id)
+        session.add(new_item)
+        session.commit()
+        flash('New Menu %s Item Successfully Created' % new_item.name)
+        return redirect(url_for('show_menu', restaurant_id=restaurant_id))
     else:
         return render_template('newmenuitem.html', restaurant_id=restaurant_id)
 
+
 # Edit a menu item
 
 
 @app.route('/restaurant/<int:restaurant_id>/menu/<int:menu_id>/edit', methods=['GET', 'POST'])
-def editMenuItem(restaurant_id, menu_id):
+def edit_menu_item(restaurant_id, menu_id):
     if 'username' not in login_session:
         return redirect('/login')
-    editedItem = session.query(MenuItem).filter_by(id=menu_id).one()
+    edited_item = session.query(MenuItem).filter_by(id=menu_id).one()
     restaurant = session.query(Restaurant).filter_by(id=restaurant_id).one()
     if login_session['user_id'] != restaurant.user_id:
-        return "<script>function myFunction() {alert('You are not authorized to edit menu items to this restaurant. Please create your own restaurant in order to edit items.');}</script><body onload='myFunction()''>"
```

Comment:
I'm not going to fix it, but there should be a standard errors page when a user goes to an invalid location.  JavaScript alerts are an ugly hack.